### PR TITLE
Cleanup document_properties.cpp

### PR DESCRIPTION
**Pull request by kroening**

### CODE REVIEW

File: src/goto-instrument/document_properties.cpp [hunk 1]
```diff
@@ -73,7 +73,7 @@ class document_propertiest
 
 void document_propertiest::strip_space(std::list<linet> &lines)
 {
-  unsigned strip=50;
+  std::size_t strip=50;
```
Comment (tautschnig) [inline src/goto-instrument/document_properties.cpp:76]: Nit pick: introduce clang-format-style whitespace while touching this line.

File: src/goto-instrument/document_properties.cpp [hunk 2]
```diff
@@ -75,11 +75,10 @@ void document_propertiest::strip_space(std::list<linet> &lines)
...
   std::size_t strip=50;
 
-  for(std::list<linet>::const_iterator it=lines.begin();
-      it!=lines.end(); it++)
+  for(const auto &line : lines)
   {
-    for(std::size_t j=0; j<strip && j<it->text.size(); j++)
-      if(it->text[j]!=' ')
+    for(std::size_t j = 0; j < strip && j < line.text.size(); j++)
+      if(line.text[j] != ' ')
```
Comment (tautschnig) [inline src/goto-instrument/document_properties.cpp:81]: I don't know what this code is supposed to do (please add documentation while doing this cleanup), but at bare minimum would like to be reassured that it does not do the same as `strip_string` from `string_utils.{h,cpp}` and also I'd like to be reassured that this code needs to live in here (and not in `string_utils.{h,cpp}`.

File: src/goto-instrument/document_properties.cpp [hunk 3]
```diff
@@ -125,14 +122,15 @@ std::string escape_html(const std::string &s)
 {
   std::string dest;
 
-  for(std::size_t i=0; i<s.size(); i++)
+  for(auto ch : s)
```
Comment (tautschnig) [inline src/goto-instrument/document_properties.cpp:125]: Shouldn't all these generic string utilities defined in this file actually be moved?

**Review by tautschnig — COMMENTED** [review]
I'm sorry, but this cleanup triggered some questions for me...

File: src/goto-instrument/document_properties.cpp (hunk 3 above)
Comment (peterschrammel) [reply src/goto-instrument/document_properties.cpp:125]: There is already a place for these in `util/string_utils.h`.

**Comment by NlightNFotis** [comment]
Hi @kroening, can we have this rebased on top of current `develop` please?

**Comment by TGWDB** [comment]
Closing due to age (no further comment on PR content), please reopen with rebase on develop if you intent to continue this work.

**Pull request closed without merging**